File: backend/Multibanca.Domain.Models/Multibanca/BBVA/Escrituracion/realizar_ep_registradas.cs
namespace Multibanca.Domain.Models.Multibanca.BBVA.Escrituracion;

public class realizar_ep_registradas
{
    public long     id { get; set; }
    public long     id_expediente { get; set; }
    public string?  id_actividad { get; set; }

    // Campos de la actividad
    public DateTime? finalizacion { get; set; }
    public string?  causal { get; set; }
    public DateTime? fecha_finalizacion { get; set; }
    public bool     confirmacion_ep_registrada { get; set; }
    public string?  observaciones { get; set; }

    // Auditoría
    public bool      is_active { get; set; }
    public bool      row_status { get; set; }
    public int       created_by { get; set; }
    public DateTime  created_date { get; set; }
    public int?      modified_by { get; set; }
    public DateTime? modified_date { get; set; }
}

File: frontend/src/features/actividades/realizar_ep_registradas/models/realizar_ep_registradas.ts
import type { Auditoria } from '@/models/Auditoria';

export interface RealizarEPRegistradas extends Auditoria {
  id: number;
  id_expediente: number;
  id_actividad: string;
  finalizacion: string | null;
  causal: string | null;
  fecha_finalizacion: string | null;
  confirmacion_ep_registrada: boolean;
  observaciones: string | null;
}

export interface DatosHeredadosEPRegistradas {
  tipo_documento?: string | null;
  numero_documento?: string | null;
  nombre_completo?: string | null;
  tipo_credito?: string | null;
  ciudad_notaria?: string | null;
  numero_notaria?: number | null;
  numero_escritura?: string | null;
  numero_boleta?: string | null;
  fecha_boleta?: string | null;
  tipo_boleta?: string | null;
  oficina_registro?: string | null;
  numero_matricula?: string | null;
}

export interface GetByExpedienteResponse {
  formulario: RealizarEPRegistradas;
  datos_heredados: DatosHeredadosEPRegistradas;
}

export const EMPTY_REALIZAR_EP_REGISTRADAS = (id_expediente: number): RealizarEPRegistradas => ({
  id: 0, id_expediente, id_actividad: 'BBVA_ESCRITURACION_REALIZAR_EP_REGISTRADAS',
  finalizacion: null, causal: null, fecha_finalizacion: null,
  confirmacion_ep_registrada: false, observaciones: null,
  is_active: true, row_status: true, created_by: 0, created_date: '', modified_by: null, modified_date: null,
});
